### PR TITLE
added comments and questions, search uppercase QUESTION to see them

### PR DESCRIPTION
**Pull request by tarronlane**

### CODE REVIEW

**Comment by tarronlane** [comment]
My comments in these files are me trying to make sense of what is going on. If I've said anything that's wrong, feel free to correct it.

File: functions/graphql/models/Stripe.js [hunk 1]
```diff
@@ -35,6 +56,13 @@ module.exports = class Stripe {
       )
     );
   }
+
+  /**
+   * Params customerId (string), planId(string), trial (bool)
+   * Returns Promise(subscription create event)
+   * QUESTION: What if they already have a subscription? would this add another subscription?
```
Comment (alexanderson1993) [inline functions/graphql/models/Stripe.js]: This should check to make sure they don't have a subscription. If they already have one, it should throw an error.

File: functions/graphql/schema/directives/auth.js [hunk 2]
```diff
@@ -19,27 +23,41 @@ module.exports.schema = gql`
...
-  // Visitor methods for nested types like fields and arguments
-  // also receive a details object that provides information about
-  // the parent and grandparent types.
+
+  // Logic to use when the directive is used on a field
+  //  - Visitor methods for nested types like fields and arguments
+  //    also receive a details object that provides information about
+  //    the parent and grandparent types.
   visitFieldDefinition(field, details) {
+    // QUESTION: If one two fields have the directive, will the first one run ensureFIeldsWrapped, mark its object as _authFieldswrapped,
```
Comment (alexanderson1993) [inline functions/graphql/schema/directives/auth.js]: I don't think that will happen. Instead, if a parent object has a role, and that role is not satisfied, then any of the children properties/objects will not be resolved. That's what that does.

However, if two properties on the same object have different roles, and one is satisfied and the other is not, then the property that is satisfied will go through just fine, while the one that is not will return null

File: functions/graphql/schema/directives/auth.js [hunk 3]
```diff
@@ -50,15 +68,18 @@ class AuthDirective extends SchemaDirectiveVisitor {
         const [data, queryArgs, context] = args;
         const { user } = context;
 
-        console.log("DATA", data);
-
+        // QUESTION: since this ensureFieldsWrapped function is called before this property is set,
```
Comment (alexanderson1993) [inline functions/graphql/schema/directives/auth.js]: I'm not entirely clear on what this question is asking, but check the above answer.

Comment (alexanderson1993) [reply functions/graphql/schema/directives/auth.js]: From the docs:

>What makes this example tricky is that the OBJECT version of the directive needs to wrap all fields of the object, even though some of those fields may be individually wrapped by @auth directives at the FIELD_DEFINITION level, and we would prefer not to rewrap resolvers if we can help it.
>One drawback of this approach is that it does not guarantee fields will be wrapped if they are added to the schema after AuthDirective is applied, and the whole getUser(context.headers.authToken) is a made-up API that would need to be fleshed out. In other words, we’ve glossed over some of the details that would be required for a production-ready implementation of this directive, though we hope the basic structure shown here inspires you to find clever solutions to the remaining problems.

**Pull request closed without merging**